WordPress/gutenberg
# (Custom)GradientPicker: Add flag to remove margins

Part of #39358 #43014

## What?

- Add a `__nextHasNoMargin` flag on CustomGradientPicker/GradientPicker to remove top and bottom margins.
- Move the CustomGradientBar component folder to be a subcomponent of CustomGradientPicker. (CustomGradientBar was an internal-only component that was not publicly exported. Its root CSS class name was `components-custom-gradient-picker__gradient-bar`, which indicates that it was never ready to be a standalone component.)
- As part of cleaning up the margins on CustomGradientBar, fix up the internal margins in DuotonePicker so it matches the ad hoc overrides that were applied when used in the block editor. (We can assume that these were always the intended correct margins.)

## Why?

For better reusability.

## How?


## Testing Instructions

1. `npm run storybook:dev`
2. Turn on the Margin Checker tool in the Storybook toolbar.
3. Check the stories for CustomGradientPicker, GradientPicker, and DuotonePicker. Compare with [trunk](https://wordpress.github.io/gutenberg).

## Screenshots or screencast 

https://user-images.githubusercontent.com/555336/185409653-a04e73c3-524c-4fe5-b32c-e15a041f1b2e.mp4
